### ISSUE TITLE
Matchit not working in visual mode. 

### ISSUE THREAD

**Issue by ReneFroger** [redguardtoo/evil-matchit#38]
Thanks for sharing this extension. Probably there is no function for visual mode. For example. 

                  <di[]v> example of sentence </div>

[] = cursor. 

In normal mode, I press `%`. Cursor goes to end of `</div>`. 
Ok, now in visual mode, cursor on same `<div>` again. I press % to visually select to end of </div> like Vim does. But this is not working in matchit.

**Comment by redguardtoo** [comment]
it's not designed for the visual mode.

it's because I already got `evil-select-items` and `evil-delete-items` in normal mode.

I assign the hotkey ",si" and ",di" for them.

say you got html as below:
```html
<div>
   hello 
</div>

<div>
  world
</div>
```

place the cursor *anywhere* at the first line, press `,si`, the hello div is selected.
press `,2si`, two div are selected.

the `,di` does the same thing. it also delete extra line returns after the end of div

I find it much more efficient than switch to visual-mode, select a region, then switch back to normal mode.

The javascript mode is also optimized for `,si`, `,di`, so you can easily select/delete  js snippets:
```javascript
$scope.hello = function() {
\}
$('#test).change(function(){
});
```

**Comment by ReneFroger** [comment]
Sorry, but that's just very inefficient. 

Try [this one](https://github.com/magnars/expand-region.el). 

And add this to your config file: 
`(define-key evil-visual-state-map (kbd "v") 'er/expand-region)`

say you got html as below:
```html

    <div>
       hello 
    </div>
```
Place the cursor anywhere at the *second* line, press v and again. The hello is selected. Press `v` again, then the selection is expanded to the two divs. 

But this was not an answer on my question, by the way. I thought the goal of Evil Matchit was to emulate the Matchit in Vim?

**Comment by redguardtoo** [comment]
I don't use visual mode too much, but I will support for visual mode.

BTW, I do use expand-region a lot (but not selecting html tags), as I I understand, in your case, press "v" cannot select *sibling* regions, right?

besides expand-region 0.10.0 with below html cannot select tags correctly  when I place cursor on the line of `hello world` and press `v`, what's you version?

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="description" content="" />
    <title>mytitle</title>
    <style type="text/css">
     .box {
       border:1px solid red;
       width: 200px;
       height: 200px;
       margin: 10px;
     }
     .red {border-color: red;}
     .green {border-color: green;}
     .blue {border-color: blue;}
    </style>
  </head>
  <body>

    <div class="box red">
      hello world
    </div>

    <div class="box blue">
      <a href="//www.gnu.org">This is <img src="//www.gnu.org/graphics/heckert_gnu.small.png" alt="FSF"/> Free Software Foundation link</a>
    </div>
    <script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/jquery/1.8.2/jquery.js"></script>
    <script type="text/javascript" src="hello.js"></script>
    <script type="text/javascript">
    </script>
  </body>
</html>
```

**Comment by redguardtoo** [comment]
4efd9e0 use % in evil-visual-mode (Chen Bin)

**Comment by ReneFroger** [comment]
I just updated Evil Matchit from Melpa now. 

And it is still not working? Maybe I'm too early to update?

**Comment by redguardtoo** [comment]
sorry, I made a mistake in evil-matchit, the %  should not be disabled by default.

fixed in 338a10c fixed: by default shortcut % is disabled (Chen Bin)

**Comment by ReneFroger** [comment]
I can confirm, so the emulation working is fine now. 

Thanks for the help, it's appreciated!
